great_tables/_formats.py

```diff
@@ -99,23 +99,25 @@ def fmt(
         The GT object is returned. This is the same object that the method is called on so that we
         can facilitate method chaining.
     """
-
-    # If a single function is supplied to `fns` then
-    # repackage that into a list as the `default` function
-    if isinstance(fns, Callable):
-        fns = FormatFns(default=fns)
-
     row_res = resolve_rows_i(self, rows)
     row_pos = [name_pos[1] for name_pos in row_res]
-
     col_res = resolve_cols_c(self, columns)
 
-    formatter = FormatInfo(fns, col_res, row_pos)
+    # If a single function is supplied to `fns` then
+    # repackage that into a list as the `default` function
+    formatter: list[FormatInfo] = []
```

Comment:
I think unfortunately GT.fmt never expects a list of functions as inputs, but uses the maybe trickily named FormatFns to mean, a dataclass that holds functions for different renderers (e.g. html, in the future latex, etc..).

It's a good point though that `GT.fmt` is an important entry point, so it's worth thinking carefully about its behavior. I think if it comes down to it, people could always pass `lambda x: formatter2(formatter1(x))`? Or use some implementation of a `compose()` function?

These docs:

```
    Parameters
    ----------
    fns
        Either a single formatting function or a named list of functions.
```

Unfortunately are copied over from R, and so named list is referring to something more like a dictionary (which we're currently using the FormatFns dataclass for)

Reply:
I'm curious about the possibility of implementing a `compose()` function. In my imagination, the code might look something like this:
```python
from functools import partial
from typing import Any, Callable
from great_tables import GT, exibble


def compose(funcs: list[Callable[Any, Any]]):
    def _compose(x: Any, funcs: list[Callable[Any, Any]]):
        for f in funcs:
            x = f(x)
        return x
    return partial(_compose, funcs=funcs)


(
    GT(exibble[["fctr"]]).fmt(
        compose([lambda x: f"_{x}_", lambda x: f"*{x}*", lambda x: f"<{x}>"]),
        columns="fctr",
    )
)
```
![image](https://github.com/posit-dev/great-tables/assets/67060418/75de906d-7bb2-4506-b7a0-937c262d4344)